### PR TITLE
Setup api for client

### PR DESCRIPTION
**Pull request by yeareater**
Setup api for client

### CODE REVIEW

File: controllers/Weathers.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	name: "Weathers",
+	run: function(req, res, next) {
+		const zoneCode = req.params.zoneCode;
+		const reqDate = req.params.date
+		if (!isValidZoneCode(zoneCode) || !isValidReqDate(reqDate)) {
+			res.send("Invalid zone code or date");
+			return;
+		}
+		const date = convertDateToDbForm(reqDate);
+		weatherUtils.loadWeather(date, zoneCode).then(function(err, weather) {
```
Comment (crowjdh) [inline controllers/Weathers.js]: **weatherUtils.loadWeather(date, zoneCode)** function will return weather only(err will be thrown to **catch** callback.

You should remove err from **function(err, weather)**

File: controllers/Weathers.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+			const refinedWeather = weatherUtils.convertWeatherToClientWeatherAPIResponse(weather);
+			res.jsonp(refinedWeather);
+		}).catch(function(err) {
+			console.log("Db error");
```
Comment (crowjdh) [inline controllers/Weathers.js]: You should respond to client's request even when there's an error.
See [here](https://github.com/Ques-Study/WeatherCalendarServer/wiki/Client-api-structure#failure) for more detail.

File: controllers/Weathers.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+const weatherUtils = require('../models/weather-utils.js');
+
+const ZONE_CODE_LENGTH = 10;
+const REQ_DATE_LENGTH = 8;
+
+module.exports = BaseController.extend({
+	name: "Weathers",
+	run: function(req, res, next) {
+		const zoneCode = req.params.zoneCode;
+		const reqDate = req.params.date
```
Comment (crowjdh) [inline controllers/Weathers.js]: Missing semi-colon here :)

File: controllers/Weathers.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+function isValidReqDate(reqDate) {
+	return typeof reqDate !== 'undefined'
+		&& reqDate.length == REQ_DATE_LENGTH;
+}
+
+function convertDateToDbForm(rawDate) {
+	const year = rawDate.slice(0,4);
+	const month = rawDate.slice(4,6);
+	const day = rawDate.slice(6,8);
+	const date = new Date(year+"-"+month+"-"+day);
```
Comment (crowjdh) [inline controllers/Weathers.js]: **new Date(year, month, day)** should work fine.

File: models/weather-api.js [hunk 5]
```diff
@@ -5,14 +5,19 @@ const FILTER_KEYS = ["hour", "day", "temp", "sky", "pty", "pop"];
 
 module.exports.FILTERING_KEYS = FILTER_KEYS;
 
-module.exports.fetch = function() {
+module.exports.fetch = function(zoneCode) {
+    var url = "http://www.kma.go.kr/wid/queryDFSRSS.jsp?zone=";
     return new Promise(function(resolve, reject) {
-        rest.get('http://www.kma.go.kr/wid/queryDFSRSS.jsp?zone=2723067100')
+        rest.get(url+zoneCode)
```
Comment (crowjdh) [inline models/weather-api.js]: I think adding whitespace in each side of **+** operator will look better.

**Review by crowjdh — CHANGES_REQUESTED** [review]
Nice work 👍 
Please check feedbacks and consider improving :)

File: controllers/Weathers.js (hunk 4 above)
Comment (DonghyunNa) [reply controllers/Weathers.js]: It makes different result.
If change code,  It needs month - 1.

**Review by DonghyunNa — APPROVED** [review]
Good job! Wuhung

File: controllers/Weathers.js (hunk 4 above)
Comment (crowjdh) [reply controllers/Weathers.js]: Hmm, it seems odd. I'll check after dinner

Comment (crowjdh) [reply controllers/Weathers.js]: Yup, you're right.

I suggest to change this line into:
const date = new Date(year, month - 1, day)

File: controllers/Weathers.js [hunk 6]
```diff
@@ -9,20 +9,24 @@ module.exports = BaseController.extend({
 	name: "Weathers",
 	run: function(req, res, next) {
 		const zoneCode = req.params.zoneCode;
-		const reqDate = req.params.date
-		if (!isValidZoneCode(zoneCode) || !isValidReqDate(reqDate)) {
-			res.send("Invalid zone code or date");
+		const reqDate = req.params.date;
+		if (!isValidZoneCode(zoneCode)) {
+			respondError(res, 400, "ZONE_CODE: " + zoneCode + " is valid");
```
Comment (crowjdh) [inline controllers/Weathers.js]: I think it should be **"invalid"**, instead of **"valid"** :)

File: controllers/Weathers.js [hunk 7]
```diff
@@ -9,20 +9,24 @@ module.exports = BaseController.extend({
...
+			return;
+		}
+		if (!isValidReqDate(reqDate)) {
+			respondError(res, 400, "DATE: " + reqDate + " is valid");
```
Comment (crowjdh) [inline controllers/Weathers.js]: Same in here, should be **"invalid"**

File: controllers/Weathers.js [hunk 8]
```diff
@@ -9,20 +9,24 @@ module.exports = BaseController.extend({
...
 			return;
 		}
 		const date = convertDateToDbForm(reqDate);
-		weatherUtils.loadWeather(date, zoneCode).then(function(err, weather) {
+		weatherUtils.loadWeather(date, zoneCode).then(function(weather) {
 			const refinedWeather = weatherUtils.convertWeatherToClientWeatherAPIResponse(weather);
 			res.jsonp(refinedWeather);
 		}).catch(function(err) {
-			console.log("Db error");
+			respondError(res, 404, "No matchin weather for ZONE_CODE: " + zoneCode + " and DATE: " + reqDate);
```
Comment (crowjdh) [inline controllers/Weathers.js]: Typo in string **"matchin"**

**Review by crowjdh — CHANGES_REQUESTED** [review]
Good work 👍 , but there're some typos ;)

**Review by crowjdh — APPROVED** [review]
Perfect :)

**Pull request merged**